game/game.py

```diff
@@ -415,46 +423,51 @@ def initialize_turn(self) -> None:
             self.compute_transit_networks()
         self.ground_planners = {}
 
-        self.blue_procurement_requests.clear()
+        if for_player:
+            self.blue_procurement_requests.clear()
         self.red_procurement_requests.clear()
 
         with logged_duration("Procurement of airlift assets"):
             self.transfers.order_airlift_assets()
         with logged_duration("Transport planning"):
             self.transfers.plan_transports()
 
-        with logged_duration("Blue mission planning"):
-            if self.settings.auto_ato_behavior is not AutoAtoBehavior.Disabled:
-                blue_planner = CoalitionMissionPlanner(self, is_player=True)
-                blue_planner.plan_missions()
-
-        with logged_duration("Red mission planning"):
-            red_planner = CoalitionMissionPlanner(self, is_player=False)
-            red_planner.plan_missions()
+        for coalition in ["Red", "Blue"]:
+            is_player = coalition == "Blue"
+            with logged_duration(f"{coalition} mission planning"):
+                if not is_player or (
+                    is_player
+                    and for_player
+                    and self.settings.auto_ato_behavior is not AutoAtoBehavior.Disabled
+                ):
```

Comment:
Conditional is easier to read IMO if this is:

```python
for player in (True, False):
    if player and (opfor_only or self.settings.auto_ato_behavior is not AutoAtoBehavior.Disabled):
        continue
    player_color = "Blue" if player else "Red"
    with logged_duration(...):
        ...
```

(it changes the logging slightly because it won't log a duration when the player planning doesn't happen, but that's an improvement anyway imo)

Reply:
the implementation with a initialize_for now solves this ugly part and i adopted your improvement with the logged duration :)